### PR TITLE
fix issues with #75

### PR DESCRIPTION
**Pull request by gatoravi**
Addresses #75

### CODE REVIEW

**Comment by yanyangfeng** [comment]
Please correct me if I'm misunderstanding the code, but it looks to me like the new block at 192 will set the cis effect window to ignore the flanking exon blocks (so whereas the splicing_exonic variants will have a window going from [=======]-------[===*]----[=====], a splicing_intronic variant will only span -------[===]*----, with * representing the variant). Similarly, the windows of intronic variants will span two introns and a central exon (the one before the variant, as in the splicing_intronic "sketch" above).

I believe what we had discussed was to keep exonic, splicing_exonic, and splicing_intronic variants so that their windows would span from the start of the left flanking exon to the end of the right flanking exon (as shown in the splicing_exonic "sketch") while changing the window for intronic variants to only spanning the intron in which the variant resides (or rather, the superset of such introns across transcripts for which the variant is "intronic"). Or am I misremembering? If not, I've proposed some changes in the code to address these concerns.

@malachig, @gatoravi

File: src/variants/variants_annotator.cc [hunk 1]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
                                                       AnnotatedVariant& variant,
                                                       uint32_t i) {
     //Check if the cis effect limits have increased.
-    if(i != 0) {
-        if(exons[i-1].start < variant.cis_effect_start) {
-            variant.cis_effect_start = exons[i-1].start;
+    if(variant.annotation == "exonic" || variant.annotation == "splicing_exonic") { //Current exon is cassette
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `if(variant.annotation == "exonic" || variant.annotation == "splicing_exonic" || variant.annotation == "splicing_intronic) {`

File: src/variants/variants_annotator.cc [hunk 2]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+        } else {
+            if(exons[exons.size() - 1].end > variant.cis_effect_end) {
+                variant.cis_effect_end = exons[exons.size() - 1].end;
+            }
         }
-    }
-    if(i != exons.size() - 1) {
-        if(exons[i+1].end > variant.cis_effect_end) {
-            variant.cis_effect_end = exons[i+1].end;
+    } else if(variant.annotation == "intronic" || variant.annotation == "splicing_intronic") { //Current exon is cassette
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `} else if(variant.annotation == "intronic") {`

File: src/variants/variants_annotator.cc [hunk 3]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+        if(i != 0) {
+            if(exons[i-1].start < variant.cis_effect_start) {
+                variant.cis_effect_start = exons[i-1].end;
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `variant.cis_effect_start = exons[i].end`

File: src/variants/variants_annotator.cc [hunk 4]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
 //Set limits on - strand
 inline
 void VariantsAnnotator::set_variant_cis_effect_limits_ns(const vector<BED>& exons,
                                                       AnnotatedVariant& variant,
                                                       uint32_t i) {
-    if(i != 0) {
-        //Check if the cis effect limits have increased.
-        if(exons[i-1].end > variant.cis_effect_end) {
-            variant.cis_effect_end = exons[i-1].end;
+    if(variant.annotation == "exonic" || variant.annotation == "splicing_exonic") { //Current exon is cassette
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `if(variant.annotation == "exonic" || variant.annotation == "splicing_exonic" || variant.annotation == "splicing_intronic) {`

File: src/variants/variants_annotator.cc [hunk 5]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+        } else {
+            if(exons[exons.size() - 1].start < variant.cis_effect_start) {
+                variant.cis_effect_start = exons[exons.size() - 1].start;
+            }
         }
-    }
-    if(i != exons.size() -1) {
-        if(exons[i+1].start < variant.cis_effect_start) {
-            variant.cis_effect_start = exons[i+1].start;
+    } else if(variant.annotation == "intronic" || variant.annotation == "splicing_intronic") { //Look at junctions that overlap intronic space
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `} else if(variant.annotation == "intronic"){`

File: src/variants/variants_annotator.cc [hunk 6]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+                variant.cis_effect_end = exons[exons.size() - 1].end;
+            }
         }
-    }
-    if(i != exons.size() - 1) {
-        if(exons[i+1].end > variant.cis_effect_end) {
-            variant.cis_effect_end = exons[i+1].end;
+    } else if(variant.annotation == "intronic" || variant.annotation == "splicing_intronic") { //Current exon is cassette
+        if(i != 0) {
+            if(exons[i-1].start < variant.cis_effect_start) {
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `if(exons[i].start < variant.cis_effect_start) {`

File: src/variants/variants_annotator.cc [hunk 7]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+            }
         }
-    }
-    if(i != exons.size() -1) {
-        if(exons[i+1].start < variant.cis_effect_start) {
-            variant.cis_effect_start = exons[i+1].start;
+    } else if(variant.annotation == "intronic" || variant.annotation == "splicing_intronic") { //Look at junctions that overlap intronic space
+        if(i != 0) {
+            //Check if the cis effect limits have increased.
+            if(exons[i-1].end > variant.cis_effect_end) {
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `if(exons[i].end > variant.cis_effect_end) {`

File: src/variants/variants_annotator.cc [hunk 8]
```diff
@@ -170,65 +170,92 @@ void VariantsAnnotator::set_variant_cis_effect_limits_ps(const vector<BED>& exon
...
+                variant.cis_effect_end = exons[i-1].start;
```
Comment (yanyangfeng) [inline src/variants/variants_annotator.cc]: `variant.cis_effect_end = exons[i].start;`

**Comment by gatoravi** [comment]
Thanks for looking this over @yanyangfeng ! How does this look, I only do this for the 'intronic' option now, the variant region is just the immediate intronic space where the variant lies. The other options are same as before, with the boundaries including the flanking exons. I also modified the code to allow partial overlap of junctions with the regions, so if either the junction start or junction end lies within the region it get's reported. Previously a junction had to be contained entirely within the window to be reported.

**Comment by yanyangfeng** [comment]
@gatoravi seems like there were a few small bugs causing errors in dealing with "intronic" variants. I pushed a commit to address these - take a look and see if you agree with my changes? As for the window overlap change - I'm wondering if we should also allow partial overlaps when the user provides a window size? I feel like it might be confusing to the user otherwise.

**Comment by gatoravi** [comment]
Hi @yanyangfeng, sorry for the slow response on this, this looks good to me. If you think partial overlaps might make it easier for the user I agree we should go for it.

**Comment by yanyangfeng** [comment]
Just want to document this here: seems the reason the coverage dropped was because some conditionals were never true so their blocks were never being executed. Somehow, though, the CI builds still passed, so it seems like there are some cases that our leaking through our tests. We may want to revisit our tests at some point to make them more robust.

**Pull request merged**